yatta/src/main.rs

```diff
@@ -631,6 +633,24 @@ fn handle_socket_message(
                                 float_titles.push(target)
                             }
                         }
+                        SocketMessage::SetWorkspace(index) => {
+                            d.set_workspace(index);
+                        }
+                        SocketMessage::MoveWindowToWorkspace(index) => {
+                            let foreground_index = d.get_foreground_window_index();
+                            d.move_window_to_workspace(index, foreground_index);
+                        }
+                        SocketMessage::MoveWindowToWorkspaceAndFollow(index) => {
+                            let foreground_index = d.get_foreground_window_index();
+                            d.move_window_to_workspace_and_follow(index, foreground_index);
+                        }
+                        SocketMessage::Stop => {
+                            let windows = desktop.get_all_windows();
+                            for mut window in windows {
```

Comment:
Can you also work this unhiding behaviour into a ctrl+c signal hook? When I'm developing I tend to run `cargo run` in the terminal and watch the logs and I have a tendency to ctrl+c out to restart the proc instead of running `yattac stop` from a different terminal, and on more than one occasion I've been left with invisible windows I had to kill with `Stop-Process` and then restart to regain visibility.

Reply:
Addressed this in my latest commit to your fork with a a ctrl-c handler that listens on a channel in a dedicated thread.